lib/galaxy/metadata/set_metadata.py

```diff
@@ -340,7 +348,15 @@ def set_meta(new_dataset_instance, file_dict):
             collect_dynamic_outputs(job_context, output_collections)
         except MaxDiscoveredFilesExceededError as e:
             final_job_state = Job.states.ERROR
-            job_messages.append(str(e))
+            job_messages.append(
+                {
+                    "type": "max_discovered_files",
+                    "desc": str(e),
+                    "code_desc": str(e),
```

Comment:
That's meant for exit code descriptions.
```suggestion
                    "code_desc": None,
```
should be fine